### DISCUSSION TITLE
Unavailability of related plugins in Vue

### DISCUSSION THREAD

**Discussion by lvhaoyu39** [katspaugh/wavesurfer.js#3122] [Q&A]
Hello, when I use the regions or timeline plugin in Vue, it doesn't work.
    wavesurferState.wavesurfer = WaveSurfer.create({
      container: waveform.value,
      height: 150,
      audioRate: 1, //音频播放速度，数值越小越慢
      hideScrollbar: false,
      autoScroll: true,
      autoCenter: true,
      minPxPerSec: 50, //音频每秒最小像素数
      cursorColor: 'red',
      waveColor: '#ecf5ff',
      progressColor: '#337ecc',
      url: new URL(url, import.meta.url).href
    })
    const wsRegion = wavesurferState.wavesurfer.registerPlugin(Region.create())
    wavesurferState.wavesurfer.registerPlugin(Timeline.create())

**Comment by katspaugh** [comment]
You omitted the most important part of the code, namely how you import the plugins.

**Reply by lvhaoyu39** [reply]
I imported it this way.

import WaveSurfer from 'https://unpkg.com/wavesurfer.js@7/dist/wavesurfer.esm.js'
import Region from 'https://unpkg.com/wavesurfer.js@7/dist/plugins/regions.esm.js'
import Timeline from 'https://unpkg.com/wavesurfer.js@7/dist/plugins/timeline.esm.js'

**Reply by katspaugh** [reply]
That's correct and should work. You should debug your app to understand what the problem is.

**Reply by lvhaoyu39** [reply]
Currently, only waveform maps are available, and the plugin has no effect.
